src/serializers/type_serializers/enum_.rs

```diff
@@ -0,0 +1,123 @@
+            serializer,
+        }
+        .into())
+    }
+}
+
+impl_py_gc_traverse!(EnumSerializer { serializer });
+
+impl TypeSerializer for EnumSerializer {
+    fn to_python(
+        &self,
+        value: &Bound<'_, PyAny>,
+        include: Option<&Bound<'_, PyAny>>,
+        exclude: Option<&Bound<'_, PyAny>>,
+        extra: &Extra,
+    ) -> PyResult<PyObject> {
+        let py = value.py();
+        if value.is_exact_instance(self.class.bind(py)) {
+            // if we're in JSON mode, we need to get the value attribute and serialize that
+            if extra.mode.is_json() {
+                let dot_value = value.getattr(intern!(py, "value"))?;
+                match self.serializer {
+                    Some(ref s) => s.to_python(&dot_value, include, exclude, extra),
+                    None => infer_to_python(&dot_value, include, exclude, extra),
+                }
+            } else {
+                // if we're not in JSON mode, we assume the value is safe to redirect directly
+                Ok(value.into_py(py))
+            }
+        } else {
+            extra.warnings.on_fallback_py(self.get_name(), value, extra)?;
+            infer_to_python(value, include, exclude, extra)
+        }
+    }
+
+    fn json_key<'a>(&self, key: &'a Bound<'_, PyAny>, extra: &Extra) -> PyResult<Cow<'a, str>> {
+        let py = key.py();
+        if key.is_exact_instance(self.class.bind(py)) {
+            let dot_value = key.getattr(intern!(py, "value"))?;
+            let k = match self.serializer {
+                Some(ref s) => s.json_key(&dot_value, extra),
+                None => infer_json_key(&dot_value, extra),
+            }?;
+            Ok(Cow::Owned(k.into_owned()))
+        } else {
+            extra.warnings.on_fallback_py(self.get_name(), key, extra)?;
+            infer_json_key(key, extra)
+        }
+    }
+
+    fn serde_serialize<S: serde::ser::Serializer>(
+        &self,
+        value: &Bound<'_, PyAny>,
+        serializer: S,
+        include: Option<&Bound<'_, PyAny>>,
+        exclude: Option<&Bound<'_, PyAny>>,
+        extra: &Extra,
+    ) -> Result<S::Ok, S::Error> {
+        if value.is_exact_instance(self.class.bind(value.py())) {
+            let dot_value = value.getattr(intern!(value.py(), "value")).map_err(py_err_se_err)?;
+            match self.serializer {
+                Some(ref s) => s.serde_serialize(&dot_value, serializer, include, exclude, extra),
+                None => infer_serialize(&dot_value, serializer, include, exclude, extra),
+            }
+        } else {
+            extra.warnings.on_fallback_ser::<S>(self.get_name(), value, extra)?;
+            infer_serialize(value, serializer, include, exclude, extra)
+        }
+    }
+
+    fn get_name(&self) -> &str {
+        Self::EXPECTED_TYPE
+    }
+
+    fn retry_with_lax_check(&self) -> bool {
+        match self.serializer {
+            Some(ref s) => s.retry_with_lax_check(),
```

Comment:
Personally I prefer the following, but I'm ok with either style (and we use both).

```suggestion
        match &self.serializer {
            Some(s) => s.retry_with_lax_check(),
```